wp-includes/class-wp-service-workers.php

```diff
@@ -63,6 +129,93 @@ public function init() {
 		do_action_ref_array( 'wp_default_service_workers', array( &$this ) );
 	}
 
+	/**
+	 * Get workbox script.
+	 *
+	 * @return string Script.
+	 */
+	public function get_workbox_script() {
+
+		$workbox_dir = 'wp-includes/js/workbox-v3.4.1/';
+
+		$script = sprintf(
+			"importScripts( %s );\n",
+			wp_json_encode( PWA_PLUGIN_URL . $workbox_dir . 'workbox-sw.js', 64 /* JSON_UNESCAPED_SLASHES */ )
+		);
+
+		$options = array(
+			'debug'            => WP_DEBUG,
+			'modulePathPrefix' => PWA_PLUGIN_URL . $workbox_dir,
+		);
+		$script .= sprintf( "workbox.setConfig( %s );\n", wp_json_encode( $options, 64 /* JSON_UNESCAPED_SLASHES */ ) );
+
+		/**
+		 * Filters whether navigation preload is enabled.
+		 *
+		 * The filtered value will be sent as the Service-Worker-Navigation-Preload header value if a truthy string.
+		 * This filter should be set to return false to disable navigation preload such as when a site is using
+		 * the app shell model.
+		 *
+		 * @param bool|string $navigation_preload Whether to use navigation preload.
+		 */
+		$navigation_preload = apply_filters( 'service_worker_navigation_preload', true ); // @todo This needs to vary between admin and backend.
+		if ( false !== $navigation_preload ) {
+			if ( is_string( $navigation_preload ) ) {
+				$script .= sprintf( "workbox.navigationPreload.enable( %s );\n", wp_json_encode( $navigation_preload ) );
+			} else {
+				$script .= "workbox.navigationPreload.enable();\n";
+			}
+		} else {
+			$script .= "/* Navigation preload disabled. */\n";
+		}
+
+		$script .= '
+if ( ! self.wp ) {
+	self.wp = {};
+}
+
+wp.serviceWorker = workbox;';
+
+		return $script;
+	}
+
+	/**
+	 * Add precaching for wp-admin and wp-includes .js and .css files.
+	 *
+	 * @todo Images are not precached.
+	 */
+	protected function precache_admin_assets() {
+
+		$routes      = array();
+		$admin_dir   = ABSPATH . 'wp-admin/';
+		$admin_files = array_merge( list_files( $admin_dir . 'css/' ), list_files( $admin_dir . 'js/' ) );
+		$inc_files   = list_files( ABSPATH . WPINC . '/js/' );
+
+		foreach ( $admin_files as $filename ) {
+			$ext = pathinfo( $filename, PATHINFO_EXTENSION );
+			if ( ! in_array( $ext, array( 'js', 'css' ), true ) ) {
+				continue;
+			}
+			$routes[] = strstr( $filename, '/wp-admin' );
+		}
+
+		foreach ( $inc_files as $filename ) {
+			$ext = pathinfo( $filename, PATHINFO_EXTENSION );
+			if ( ! in_array( $ext, array( 'js', 'css' ), true ) ) {
+				continue;
+			}
+
+			// @todo This will cache both min.js and .js, however, not all the files have .min.js. Is it OK to cache all the files?
+			$routes[] = strstr( $filename, '/wp-includes' );
```

Comment:
Should we try to precache only the files which are visibly minified (ending with `.min.js`) or don't have a minified version separated?

Reply:
If there are two versions of a given URL, cache the min version. Otherwise, if there is no min version then cache the regular.

For example, jQuery is bundled with core and it is pre-minified so there is no minified version. The `jquery-migrate` script, however, does have a minified version:

https://github.com/WordPress/wordpress-develop/blob/c52e7c79b2fd6381ba89fb8c40624e6621284743/src/wp-includes/script-loader.php#L211-L212

Basically look for the presence of the `$suffix` variable in the registered URLs.